ethereum-optimism/optimism
# Develop -> Master

- test(ctb): Add a fuzz test for large deposit gaps
- feat: bindings script for cicleci breaking prevention
- feat: bridge event tests
- tests: erc20 events
- op-node: Document metrics
- specs: cleanup bedrock derivation specs
- Add support for stopping / starting sequencer from admin API
- Use a reader instead of offsets when parsing the `SystemConfig` update event in `op-node`
- Inline read func; ignore possible errors from `Read` due to length check
- Change `readIntoWord` -> `readWord`
- Make `readWord` return a pointer to a fixed-length byte arr
- fix: expected events and calls :test_tube:
- Return value rather than pointer
- Move `uint64` padding bytes into a named var
- Move `address` padding bytes into a named var
- nit: var formatting
- Fix copy pasta error; :broom:
- specs: derivation doc review suggestion fixes
- specs: fix text about finalization to insert missing proof examples
- specs: derivation spec lint fix
- Use proto's suggested error handling method for `readWord`
- op-chain-ops: `Withdrawal` type use `hexutil.Bytes`
- op-chain-ops: add getter for cdm
- tests: expand l1 standard bridge testing :test_tube:
- Move helper var declarations to the package level
- op-chain-ops: add in json struct tags to cdm
- tests: L2 bridge receiver :test_tube:
- ctp: update AttestationStation to 1.1.0
- Receive latest block hash from admin_stopSequencer Pass block hash to admin_startSequencer
- tests: L2 bridge withdrawal calls and events :test_tube:
- tests: L2 bridge withdraw to calls and events :test_tube:
- Add sequencer.stopped flag that can be used to start a node in sequencer mode that doesn't produce blocks
- tests: L2 bridge fully tested
- fix: ci nits :bug:
- Update CODEOWNERS
- specs: op-node fix broken g-finalized-l2-head reference
- ctb: Fix gas snapshot
- Revert "ctb: Fix gas snapshot"
- ctb: Fix gas snapshot
- fix: gas snapshot with new foundry ci version bump
- Compile solc artifacts before spacer validation
- fix: ci nits - let's see if this works :rocket:
- fix: storage snapshot
- op-node: attributes builder interface
- fix: storage snapshot
- fix: storage snapshot
- fix: storage snapshot
- fix: storage snapshot
- fix: storage snapshot
- contracts-bedrock: Fix deploy-config type `channelTimeout` description
- specs: Fix channel timeout description
- tech docs: Add ToB Bedrock Updates report
- tech docs: More consistent file naming
- op-node: fix pprof missing handler, use op-service pprof util instead
- :bug: fix: Allow unknown option in base service v2
- :memo: changeset
- Fix typo
- fix(ctb): ERC20 bridge events
- contracts-bedrock: update semver
- op-bindings: regenerate
- contracts-bedrock: update check-l2 task
- contracts-bedrock: remove virtual receive impl
- contracts-bedrock: update semver test
- op-chain-ops: add comment to lwd -> xdm
- ctp: simplify station
- chore(ctb): Update gas snapshot
- docs: Add Bedrock and Periphery Audit
- op-chain-ops: Preserve predeploy nonces
- INF-22 implements op-signer client
- INF-22 integrate op-signer client with op-batcher
- chore(2step): copy over scaffolding from fault-detector
- chore(ops): Add 2 step mon to Dockerfile.packages
- feat(ci): Add 2 step monitor to depcheck job
- l2geth: default gascap
- op-proposer: Cleanup
- gas snapshot
- op-bindings: Check for duplicate artifacts
- op-chain-ops: Use precalculated mint events
- INF-22 integrate op-signer client with op-proposer
- proxyd: Add sender-based rate limiter
- feat: op-node docs
